apps/explorer/src/app/routes/router-config.tsx

```diff
@@ -128,80 +197,131 @@ const validators: Route[] = flags.validators
   ? [
       {
         path: Routes.VALIDATORS,
-        name: t('Validators'),
-        text: t('Validators'),
+        handle: {
+          name: t('Validators'),
+          text: t('Validators'),
+          breadcrumb: () => (
+            <Link to={Routes.VALIDATORS}>{t('Validators')}</Link>
+          ),
+        },
         element: <ValidatorsPage />,
       },
     ]
   : [];
 
-const routerConfig: Route[] = [
+const linkTo = (...segments: (string | undefined)[]) =>
+  compact(segments).join('/');
+
+export const routerConfig: Route[] = [
   {
     path: Routes.HOME,
-    name: t('Home'),
-    text: t('Home'),
-    element: <Home />,
-    index: true,
-  },
-  {
-    path: Routes.TX,
-    name: t('Txs'),
-    text: t('Transactions'),
-    element: <Txs />,
+    element: <Layout />,
+    handle: {
+      name: t('Home'),
+      text: t('Home'),
+      breadcrumb: () => <Link to={Routes.HOME}>{t('Home')}</Link>,
+    },
+    errorElement: <ErrorBoundary />,
     children: [
-      {
-        path: 'pending',
-        element: <PendingTxs />,
-      },
-      {
-        path: ':txHash',
-        element: <Tx />,
-      },
       {
         index: true,
-        element: <TxsList />,
-      },
-    ],
-  },
-  {
-    path: Routes.BLOCKS,
-    name: t('Blocks'),
-    text: t('Blocks'),
-    element: <BlockPage />,
-    children: [
-      {
-        index: true,
-        element: <Blocks />,
+        element: <Home />,
       },
       {
-        path: ':block',
-        element: <Block />,
+        path: Routes.TX,
+        handle: {
+          name: t('Txs'),
+          text: t('Transactions'),
+          breadcrumb: () => <Link to={Routes.TX}>{t('Transactions')}</Link>,
+        },
+        children: [
+          {
+            path: 'pending',
+            element: <PendingTxs />,
+            handle: {
+              breadcrumb: () => (
+                <Link to={linkTo(Routes.TX, 'pending')}>
+                  {t('Pending transactions')}
+                </Link>
+              ),
+            },
+          },
+          {
+            path: ':txHash',
+            element: <Tx />,
+            handle: {
+              breadcrumb: (params: Params<string>) => (
+                <Link to={linkTo(Routes.TX, params.txHash)}>
+                  {params.txHash}
+                </Link>
+              ),
+            },
+          },
```

Comment:
**praise (non-blocking)**: I like this pattern for adding breadcrumbs, nice and easy to follow¹ 👍 

¹ Pun unintentional